lib/loomio/loomio_handler.rb

```diff
@@ -0,0 +1,73 @@
+require 'httparty'
+
+class LoomioHandler
+  include HTTParty
+  @@base_uri = 'https://talk.theborderland.se/'
+  @@sessions_uri = 'api/v1/sessions'
+  @@comments_uri = 'api/v1/comments'
+  @@threads_uri = 'api/v1/discussions'
+
+  # login using username and password and set cookies
+  # for all the other requests within LoomioHandler
+  def initialize(username, password)
+    user = {user:{email:username,password:password}}
+    security_response = self.class.post(@@base_uri + @@sessions_uri, body: user)
+    puts(security_response.value) # raises error if the request failed
+
+    # parse and set security cookies based on the post call
+    @security_cookies = self.parse_set_cookie(security_response.headers['set-cookie'])
+    self.class.default_cookies.add_cookies(@security_cookies)
+
+    # set headers used for all requests
+    @headers = {
+      'Content-Type': 'application/json'
+    }
+  end
+
+  # create a thread (discussion) in loomio
+  def new_thread(name, description='')
+    response = self.class.post(
+      @@base_uri + @@threads_uri, 
+      body: {"discussion":{"title":name,"description":description,"group_id":1259,"private":false,"forked_event_ids":[],"document_ids":[]}}.to_json,
+      headers: @headers
+    )
+    puts(response.value) # raises error if the request failed
+    return response
+  end
+
+  # post a comment to loomio based on discussion_id
+  def new_comment(name, discussion_id)
+    response = self.class.post(
+      @@base_uri + @@comments_uri, 
+      body: {"comment":{"body":name,"discussion_id":discussion_id,"document_ids":[]}}.to_json,
+      headers: @headers
+    )
+  end
+
+  # parse cookies coming from a set-cookies header of a response
+  def parse_set_cookie(all_cookies_string)
+    cookies = Hash.new
```

Comment:
I know this is the SO-accepted answer, but I'd have a look to see if
```
self.class.default_cookies.add_cookies CGI::Cookie.parse(security_response.headers['set-cookie'])
```
did what you needed first.
https://ruby-doc.org/stdlib-1.9.3/libdoc/cgi/rdoc/CGI/Cookie.html